src/sortStudents.ts

```diff
@@ -1,16 +1,52 @@
-
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const direction: -1 | 1 = order === 'asc' ? 1 : -1;
+
+  return [...students]
+    .sort(({ [sortBy]: paramA }, { [sortBy]: paramB }): number => {
+      let diff: number = 0;
+
+      if (typeof paramA === 'number' && typeof paramB === 'number') {
+        diff = paramA - paramB;
+      }
+
+      if (typeof paramA === 'string' && typeof paramB === 'string') {
+        diff = paramA.localeCompare(paramB);
+      }
+
+      if (typeof paramA === 'boolean' && typeof paramB === 'boolean') {
+        diff = Number(paramA) - Number(paramB);
+      }
```

Comment:
you can combine it. But this condition is hard to read, so maybe better use SortType here )

```suggestion
      if (typeof paramA === 'string' && typeof paramB === 'string') {
        diff = paramA.localeCompare(paramB);
      }

      if ((typeof paramA === 'boolean' && typeof paramB === 'boolean')
        || (typeof paramA === 'number' && typeof paramB === 'number')
      ) {
        diff = Number(paramA) - Number(paramB);
      }
```